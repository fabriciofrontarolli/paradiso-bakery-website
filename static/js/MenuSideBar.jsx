import React, { Component } from 'react';

import MenuSideBarButton from './MenuSideBarButton.jsx';

class MenuSideBar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      menuitems: ['Breakfast', 'Boxed Lunches', 'Entree Salads', 'Gourmet Trays and Appetizers', 'Main Selections', 'Pizzas']
    };
  }

  render() {
    const {menuitems} = this.state;
    const menuitemlist = menuitems.map( (item) => <MenuSideBarButton menuitem={item} />);
    return (
      <aside className="menu_sidebar__container">
        {menuitemlist}
        <div className="bottom-sidebar-space"></div>
      </aside>
    );
  }

}

export default MenuSideBar;
