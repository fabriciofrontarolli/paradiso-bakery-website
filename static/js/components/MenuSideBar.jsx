import React, { Component } from 'react';

import MenuSideBarButton from './MenuSideBarButton.jsx';

import * as constants from '../constants/constants.js';

class MenuSideBar extends Component {

  constructor(props) {
    super(props);
    this.state = {
      menuitems: constants.RECIPE_SECTIONS
    };
  }

  render() {
    const {menuitems} = this.state;
    const keys = menuitems.map((val, index, arr) => {
      return `${val.split(' ')[0].toLowerCase()}${index}`;
    });
    const menuitemlist = menuitems.map( (item, index) => <MenuSideBarButton itemArea={this.props.itemArea} menuitem={item} key={keys[index]} />);
    return (
      <ul className="pure-menu custom-restricted-width menu_sidebar__container">
        {menuitemlist}
      </ul>
    );
  }

}

export default MenuSideBar;
