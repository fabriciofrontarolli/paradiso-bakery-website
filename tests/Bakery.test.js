import renderer from 'react-test-renderer';
import React from 'react';

import MenuSideBar from '../static/js/components/MenuSideBar.jsx';

test('Show Links in Main bakery site', () => {
  const menuitems = ['Breakfast', 'Breakfast Ala Carte','Boxed Lunches', 'Entree Salads', 'Gourmet Trays and Appetizers', 'Main Selections', 'Pizzas'];
  const tree = renderer.create(
    <MenuSideBar menuitems={menuitems} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
