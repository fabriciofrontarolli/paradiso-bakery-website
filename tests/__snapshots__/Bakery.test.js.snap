exports[`test Show Links in Main bakery site 1`] = `
<aside
  className="menu_sidebar__container">
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Breakfast
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Breakfast Ala Carte
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Boxed Lunches
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Entree Salads
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Gourmet Trays and Appetizers
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Main Selections
  </button>
  <button
    className="menu_sidebar_button__container"
    onClick={[Function]}>
    Pizzas
  </button>
  <div
    className="bottom-sidebar-space" />
</aside>
`;
